mobilecoind/api/proto/mobilecoind_api.proto

```diff
@@ -611,6 +628,46 @@ message GenerateBurnRedemptionTxResponse {
     TxProposal tx_proposal = 1;
 }
 
+// Generate a simple swap proposal. The result is a signed contingent input
+// which trades one currency for another and is suitable for use with the deqs.
+// (This API is restrictive and doesn't let you build more complex SCIs.)
+message GenerateSwapRequest {
+    // Monitor id sending the funds.
+    bytes sender_monitor_id = 1;
+
+    // Subaddress to return change to.
+    uint64 change_subaddress = 2;
+
+    // A specific input, whose value will be offered in full by the swap.
+    //
+    // You may need to conduct a self-spend first to find an input of exactly
+    // the correct value before using this API if none of your inputs match
+    // the offer you want to make.
+    UnspentTxOut input = 3;
+
+    // The u64 value we are asking for in exchange for our input
+    uint64 ask_value = 4;
```

Comment:
Might want to rename to counter_ instead of ask_ if we want the same terminology the deqs uses

Reply:
> maybe we have a list of MOB/EUSD quotes and a list of EUSD/MOB quotes
Thats how its currently set up

Reply:
okay -- i will change it to "counter" then